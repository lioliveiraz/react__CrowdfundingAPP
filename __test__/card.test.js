import { shallow } from 'enzyme';
import ProjectCard from '../Components/ProjectList/Project/Card';

describe("Project Card", () => {
    let wrapper;


    describe('<Card> rendering', () => {
        beforeEach(() => {
            const mockProjectObject = {
                id: 1,
                name: " name",
                goal: 1,
                donated: 1,
                creator: "creator",
                image: "image",
                description: "description"
            };
            wrapper = shallow(
                <ProjectCard project={mockProjectObject} />
            );

        });
        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
        it('renders tags correctly', () => {
            expect(wrapper.find('img')).toHaveLength(1);
            expect(wrapper.find('h3')).toHaveLength(1);

        });
        it('renders elements with  props', () => {
            expect(wrapper.containsMatchingElement(<img alt="project name  name" src="image" />)).toBeTruthy();
            expect(wrapper.containsMatchingElement(<h3>name</h3>)).toBeTruthy();

        });

    });



});