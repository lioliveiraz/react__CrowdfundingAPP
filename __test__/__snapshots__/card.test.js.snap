// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Card <Card> rendering renders correctly 1`] = `ShallowWrapper {}`;
